include/ap/argument_parser.hpp

```diff
@@ -221,43 +421,62 @@ class optional_argument : public argument_interface {
     friend class ::ap::argument_parser;
 
 private:
+    /**
+     * @brief Sets the value for the optional argument.
+     * @param value The value to set.
+     * @return Reference to the optional_argument.
+     */
     inline optional_argument& value(const std::any& value) override {
         this->_value = value;
         return *this;
     }
 
+    //! @copydoc argument_interface::is_positional
+    [[nodiscard]] inline bool is_positional() const override {
+        return not this->_optional;
+    }
+
+    //! @copydoc argument_interface::is_optional
+    [[nodiscard]] inline bool is_optional() const override {
+        return this->_optional;
+    }
+
+    //! @copydoc argument_interface::has_value
     [[nodiscard]] inline bool has_value() const override {
         return this->_value.has_value();
     }
 
+    //! @copydoc argument_interface::value
     [[nodiscard]] inline const std::any& value() const override {
         return this->_value;
     }
 
+    //! @copydoc argument_interface::name
     [[nodiscard]] inline const argument_name& name() const override {
         return this->_name;
     }
 
+    //! @copydoc argument_interface::required
     [[nodiscard]] inline bool required() const override {
         return this->_required;
     }
 
+    //! @copydoc argument_interface::help
     [[nodiscard]] inline const std::optional<std::string_view>& help() const override {
         return this->_help_msg;
     }
 
+    //! @copydoc argument_interface::default_value
     [[nodiscard]] inline const std::any& default_value() const override {
         return this->_default_value;
     }
 
-    const bool _optional = true;
-    const argument_name _name;
-
-    std::any _value;
-
-    bool _required = false;
-    std::optional<std::string_view> _help_msg;
-    std::any _default_value;
+    const bool _optional = true;                ///< Indicates if the optional argument is optional.
+    const argument_name _name;                  ///< Name (and optional short name) of the optional argument.
+    std::any _value;                            ///< Value of the optional argument.
+    bool _required = false;                     ///< Indicates if the optional argument is required.
+    std::optional<std::string_view> _help_msg;  ///< Optional help message for the optional argument.
+    std::any _default_value;                    ///< Default value for the optional argument.
 
 #ifdef AP_TESTING
```

Comment:
Is this `#ifdef` block not finished or am I trippin'?
Check whether this is a result of resolving merge conflicts